Assets/PackageStagingArea/eu.netherlands3d.cameras/Runtime/Scripts/Cameras/Compass.cs

```diff
@@ -0,0 +1,130 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using DG.Tweening;
+using GG.Extensions;
+using Netherlands3D.Events;
+using UnityEngine;
+using UnityEngine.InputSystem;
+using UnityEngine.UI;
+
+namespace Netherlands3D.Twin.PackageStagingArea.eu.netherlands3d.cameras.Runtime.Scripts.Cameras
+{
+    public class Compass : MonoBehaviour
+    {
+        public RectTransform arrowTransform;
+        public Color NorthColor;
+        
+        private Camera cameraComponent;
+        private FreeCamera freeCamera;
+        private Transform cameraTransform;
+        private bool cancelingAnimation = false;
+        private Image arrowImage;
+        private Color arrowColor;
+
+        private const float directionMargin = 0.1f;
+        private const float northAngleMargin = 1.0f;
+
+        [Header("Animation tweaks")]
+        [SerializeField] private float snapSpeed = 10f;
+
+        [Header("Input")][SerializeField] private InputActionAsset inputActionAsset;
+        private InputAction rotateAction;
+        private InputActionMap cameraActionMap;
+
+        private void Awake()
+        {
+            cameraComponent = GetComponent<Camera>();
+            freeCamera = GetComponent<FreeCamera>();
+            cameraTransform = cameraComponent.transform;
+
+            arrowImage = arrowTransform.GetComponent<Image>();
+            arrowColor = arrowImage.color;
+
+            //when user rotates, cancel the animation if active
+            cameraActionMap = inputActionAsset.FindActionMap("Camera");
+            rotateAction = cameraActionMap.FindAction("RotateModifier");
+            rotateAction.started += CancelAnimation;
+        }
+
+        private void Update()
+        {
+            arrowTransform.SetRotationZ(cameraTransform.transform.eulerAngles.y);
```

Comment:
Kan dit ook event-gedreven i.p.v. elke update? Door bijvoorbeeld naar de RotateInput scriptable object event te luisteren? En ipv deze berekenen door naar de cameraTransform.eulerAngles.y te kijken? Want als ik mij niet vergis dan is de y euler angle altijd gelijk aan de Yaw en is 0 altijd gelijk aan het noorden?

Reply:
dat klopt zo had ik het eerst ook, maar kreeg wat problemen met zomaar missende transforms uit een input action callback systeem. Nu moet er een null check op staan. 
ja klopt eulerY==yaw.

Reply:
dit is inmiddels gedaan met DOTween en nu volledig opgelost